### PR TITLE
Fix undefined behavior in the editor wasm bridge

### PR DESCRIPTION
**Pull request by mfish33**
This should remove the possible undefined behavior, described by @bjorn3, that was introduced in #379.

### CODE REVIEW

**Comment by Keavon** [comment]
@bjorn3 would you mind reviewing this change? Thanks 😊

**Comment by bjorn3** [comment]
I think this indeed fixes the issue.

**Comment by Keavon** [comment]
Awesome, thanks for taking a look. I'll give my approval. Waiting on @TrueDoctor but we could merge early if needed.

**Review by Keavon — APPROVED** [review]

File: frontend/wasm/src/api.rs [hunk 1]
```diff
@@ -31,8 +31,11 @@ pub struct Editor {
...
 	pub fn new(handle_response: js_sys::Function) -> Editor {
+		let editor_id = generate_uuid();
+		let editor = editor::Editor::new();
+		EDITOR_INSTANCES.with(|instances| instances.borrow_mut().insert(editor_id, editor));
 		Editor {
-			editor: UnsafeCell::new(editor::Editor::new()),
+			editor_id,
 			instance_received_crashed: Cell::new(false),
 			handle_response,
 		}
```
Comment (TrueDoctor) [inline frontend/wasm/src/api.rs:41]: Could we make this naming a little less confusing? Having two `Editor`s is a bit confusing, had to read the function twice to understand what was going on.

**Review by TrueDoctor — COMMENTED** [review]
this seems a little bit weird to me, if we already have a hashmap of editors if would expect the dispatch function to just take an editor id. Then the wasm::Editor would be stored in the hashmap directly

**Comment by mfish33** [comment]
> this seems a little bit weird to me, if we already have a hashmap of editors if would expect the dispatch function to just take an editor id. Then the wasm::Editor would be stored in the hashmap directly

Can you explain further on this? The point of the hashmap behind the thread-local is to be able to modify an instance of an editor with a shared reference rather than a mutable one

**Comment by TrueDoctor** [comment]
> Can you explain further on this? The point of the hashmap behind the thread-local is to be able to modify an instance of an editor with a shared reference rather than a mutable one. I also updated the naming of Editor to JsEditorHandle to hopefully make it less confusing.

we are using a hashmap to support multiple editors right?
But we still keep one global instance of the JsEditorHandle.
I feel like it would make sense to go all the way and store these editor handles in the hashmap itself, otherwise I don't really see the point of adding a hashmap

**Review by TrueDoctor — APPROVED** [review]
Missed the last pr getting merged, this looks all good

**Comment by mfish33** [comment]
After Discussion @TrueDoctor and myself agree that this PR is ready to go

**Pull request merged**